falcon/routing/compiled.py

```diff
@@ -43,8 +44,12 @@ def add_route(self, uri_template, method_map, resource):
         """Adds a route between URI path template and resource."""
         # Can't start with a number, since these eventually get passed as
         # args to on_* responders
-        if re.search('{\d', uri_template):
-            raise ValueError('Field names may not start with a digit.')
+
+        fields = re.findall('{(.+?)}', uri_template)
```

Comment:
Should we also look for the empty string? E.g., given a path template of `'/some/{}/template'`we would raise an error?